src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -3282,7 +3281,8 @@ ByteVector fromBooleanArray(VectorSpecies<Byte> species,
      *         or {@code offset+N*1 >= ms.byteSize()}
      *         for any lane {@code N} in the vector
      * @throws IllegalArgumentException if the memory segment is a heap segment that is
-     *         not backed by a {@code byte[]} array.
+     *         not backed by a {@code byte[]} array and if access to the backing array
```

Comment:
This seems backwards - e.g. a ByteVector requires an alignment of 1. So I would expect to be able to pass _any_ segment?